DataAccess/Concrete/EntityFramework/EfProductRepository.cs

```diff
@@ -10,28 +10,28 @@ namespace WebNetSample.DataAccess.Concrete.EntityFramework;
 
 public class EfProductRepository : EfEntityRepositoryBase<Product, WebNetSampleDBContext>, IProductRepository
 {
+    private readonly WebNetSampleDBContext context;
+
     public EfProductRepository(WebNetSampleDBContext databaseContext) : base(databaseContext)
     {
+        context = databaseContext;
     }
 
     public async Task<List<ProductDetailDto>> GetProductDetailsAsync(Expression<Func<Product, bool>> filter = null)
     {
-        using (var context = new WebNetSampleDBContext())
-        {
-            var result = from product in context.Products
-                         join category in context.Categories on product.CategoryId equals category.Id
-                         join supplier in context.Suppliers on product.SupplierId equals supplier.Id
-                         select new ProductDetailDto()
-                         {
-                             Id = product.Id,
-                             Name = product.Name,
-                             Price = product.Price,
-                             ImageUrl = product.ImageUrl,
-                             CategoryName = category.Name,
-                             SupplierName = supplier.Name
-                         };
+        var result = from product in context.Products
+                     join category in context.Categories on product.CategoryId equals category.Id
+                     join supplier in context.Suppliers on product.SupplierId equals supplier.Id
+                     select new ProductDetailDto()
```

Comment:
Not really agree with your approach here: this is a data access layer, not business or front. DTO/ViewModels and other similar things should be only applied definitely not on DAL. In your case you should return entity `Product` and perform mapping on business layer. It will not violate N-layer architecture.